include/net/xia_ether.h

```diff
@@ -240,6 +243,105 @@ static int attach_neigh_addr_to_fib_entry(struct fib_xid_ether_main
 	return 0;
 }
 
+/* Template struct containing hdr manipulation funcs. */
+struct xia_header_ops {
+	/* Creates a new L2 header. */
+	int (*create) (struct sk_buff *skb, struct net_device *dev,
+			   unsigned short type, const void *daddr,
+			   const void *saddr, unsigned int len);
+
+	/* Extracts hrdw address from pkt. */
+	int (*parse)(const struct sk_buff *skb, unsigned char *haddr);
+
+	/* Creates a cached hdr and alocates to a mfxid. */
+	int (*cache)(const struct fib_xid_ether_main *mfxid, struct hh_cache *hh, __be16 type);
+
+	/* Updates the cached hdr on change of address
+	 * type : 1 -> destionation addr
+	 *        0 -> source addr
+	 */
+	void (*cache_update)(struct hh_cache *hh, const struct net_device *dev,
```

Comment:
Before dropping `struct xia_header_ops` (see another comment for details on this), move the above comment to `xia_ether_header_cache_update()`.